router/http.go

```diff
@@ -116,6 +116,10 @@ func (s *HTTPListener) startSync(ctx context.Context) error {
 	}
 }
 
+// Testing hooks
+var preResync func()
+var postResync func(startc <-chan struct{})
```

Comment:
I'd move these to the listener struct so that multiple tests can be run concurrently.